### PR TITLE
Update react/dns to use v1 or v0

### PR DESCRIPTION
**Pull request by nenad**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | no
| Documentation   | no
| License         | MIT

#### What's in this PR?

Uses either the v1 or v0 of the `react/dns` package, depending on constraints.

#### Why?

Allows projects to update dependencies that depend on `react/dns` to newer versions.

#### Checklist

- [ ] Updated CHANGELOG.md to describe BC breaks / deprecations | new feature | bugfix
- [ ] Documentation pull request created (if not simply a bugfix)

#### To Do

- [ ] If the PR is not complete but you want to discuss the approach, list what remains to be done here

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,7 +7,6 @@ cache:
         - $HOME/.composer/cache/files
 
 php:
-    - 5.5
```
Comment (xabbuh) [inline .travis.yml]: IMO this change should be discussed in another PR as it is not related to allowing `react/dns` 1.x.

Comment (nenad) [reply .travis.yml]: Resolved in #37

File: composer.json [hunk 2]
```diff
@@ -14,7 +14,7 @@
         "php": "^5.5 || ^7.0",
         "php-http/httplug": "^1.0",
         "react/http-client": "~0.5.9",
-        "react/dns": "^0.4.15",
+        "react/dns": "^1.0|^0.4",
```
Comment (dbu) [inline composer.json]: why do you allow 0.4.0 again? #33 increased to 0.4.15 and from the PR description i would assume that lower versions have an issue.

Comment (nenad) [reply composer.json]: `^0.4` would pick the latest 0.4.x version that is available. This is in the composer docs for pre-1.0 versions:
> For pre-1.0 versions it also acts with safety in mind and treats `^0.3` as `>=0.3.0 <0.4.0`.

If you would like me to revert the change, I will.

Comment (dbu) [reply composer.json]: yes, if you just upgrade everything, composer will install the latest 0.4 (or 1.0 now). however, if a user for some reason decided to require 0.4.13 in their project, they would get a conflict because we require 0.4.15 or better. with your change, they would install a version that is incompatible with our code. for this reason, we have the `--prefer-lowest` build in [travis](https://github.com/php-http/react-adapter/blob/e3dc672bc5211fb20556bdacd526eecc26fb1620/.travis.yml#L30) which should fail now. (looking at it, it seems to have installed 0.4.13 and not failed - i guess we could theoretically do `^0.4.13` but probably not worth it)

i would make this `^1.0|^0.4.15`.

Comment (nenad) [reply composer.json]: Gotcha :+1: I'll update the requirements!

**Comment by dbu** [comment]
thanks a lot @nenad !
